packages/theme/src/cli/utilities/asset-ignore.ts

```diff
@@ -56,9 +56,17 @@ function matchGlob(key: string, pattern: string) {
   // When the the standard match fails and the pattern includes '/*.', we
   // replace '/*.' with '/**/*.' to emulate Shopify CLI 2.x behavior, as it was
   // based on 'File.fnmatch'.
-  if (pattern.includes('/*.')) {
-    outputWarn(`The pattern ${pattern} is unsafe, please use a valid regex or glob.`)
-    return originalMatchGlob(key, pattern.replace('/*.', '/**/*.'))
+  if (pattern.includes('/*.') && !pattern.includes('/**/*.')) {
+    const newPatternMatch = originalMatchGlob(key, pattern.replace('/*.', '/**/*.'))
+    if (newPatternMatch) {
```

Comment:
Personally, I don't think we should adopt pattern matching or not matching as our criteria for showing the warning, as the pattern is still used when it's not being matched.

What do you think about moving this warning to the `applyIgnoreFilters` level? Then, we'd display this warning once per pattern (instead of per pattern usage), and we'd still inform users about this :)

Reply:
https://github.com/Shopify/cli/pull/3669#discussion_r1555374641